python/tvm/meta_schedule/relay_integration.py

```diff
@@ -45,8 +45,7 @@
 from .tune_context import TuneContext
 from .utils import fork_seed
 
-if TYPE_CHECKING:
-    from tvm import relay
+from tvm import relay
```

Comment:
By design, MetaSchedule should not depend on relay. Let's keep this line as it is

Reply:
Ok. But there is two things:
1. Meta-scheduler potentially can depend on relay if TYPE_CHECKING=True
2. This line fixed the issue, I cannot simply skip it
Of course I can use `from tvm.topi.x86.utils import target_has_vnni`, but it does not change anything due to topi depends on relay.
May be do you know another way?

Reply:
> Meta-scheduler potentially can depend on relay if TYPE_CHECKING=True

This is a python's type checking mechanism, so you may assume "TYPE_CHECKING" is always "False" when the program is executed: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

> due to topi depends on relay

It is a sad fact and potentially a design issue. Ideally, TOPI should never depend on Relay in the first place, because it does not need to: TOPI + TE + TIR could do code generation without much problem at all. Let's start do things in a right way from new modules like MetaSchedule or Relax.

> Of course I can use from `tvm.topi.x86.utils import target_has_vnni`, but it does not change anything

We could import locally in a method in `relay_integration.py`, so that the importing won't happen until users are sure that they want to import Relay.

Reply:
Thank you @junrushao for your answer. Yes, we can import something locally (for example, during space generator initializing), but I do not think it is good solution as result. So, I tag WIP for this PR and waiting for ideas from [issue](https://github.com/apache/tvm/issues/13604) and/or [discussion](https://discuss.tvm.apache.org/t/failure-of-registration-of-global-function-in-tvm-python-api/14093). In one case, if topi-relay dependences will be avoided, I can import needed func from topi on meta-scheduler side. In another case, if global functions will be separated (for example, from topi.x86.topi), I use them without dependences on relay or topi.
Just now I use fix on my test script side, but I think it should be fixed as result in some correct way.